pre-commit

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+STAGED_GO_FILES=$(git diff --cached --name-only | grep ".go$" | grep -v "mock")
+
+for FILE in ${STAGED_GO_FILES}
+do
+    gofmt -w -s "${FILE}"
+    goimports -w "${FILE}"
+    git add "${FILE}"
+done
+
+if [[ -n "${STAGED_GO_FILES}" ]]; then
+    set -o errexit
+    go test --tags="unit integration" -race ./internal...
```

Comment:
[q] For my own knowledge, do we have any tests yet?

Reply:
Actually looking at the codebase, we don't have a `internal` as well as a dual root structure with cfn + cdk folders.

I don't think this line makes sense yet until we add more testing strucutre.

Reply:
sorry I have actually committed the wrong file

Reply:
updated